### ISSUE TITLE
Ember serve fails: "Cannot read property 'map' of undefined"

### ISSUE THREAD

**Issue by ppegusii** [shipshapecode/ember-flatpickr#493]
I'm trying to create an addon with ember-flatpickr as one of its dependencies. When I add ember-flatpickr as a dependency in the addon, the project consuming my addon fails on ember serve. However, if I install ember-flatpickr directly in the consuming project and remove it from package.json in the addon, it works fine. Any idea what I'm doing wrong?

Thanks very much for your help!

**Comment by rwwagner90** [comment]
Likely you need to call this line in your addon https://github.com/shipshapecode/ember-flatpickr/blob/master/index.js#L53

**Comment by ppegusii** [comment]
Thanks for your quick reply. Unfortunately, I get the same error. Here's the index.js of my addon:

```
'use strict';

module.exports = {
  name: require('./package').name,

  included(){
    this._super.included.apply(this, arguments);
  }
};
```
I'm using version 2.14.3 of ember-flatpickr and the ember version of my addon and consuming project is 3.4. [Here's](https://github.com/shipshapecode/ember-flatpickr/blob/a28b6ab2246f1aa8d2651cc4425b201591c6983e/index.js#L10) the exact line that's throwing the error.

**Comment by rwwagner90** [comment]
Right, the `included` hook of this addon has to run before the one in your addon, since it defines `this.locales`. You can also try adding this to your package.json:

```json
 "ember-addon": {
    "after": [
      "ember-flatpickr"
    ]
  },
```

This should theoretically ensure ember-flatpickr runs first.

**Comment by ppegusii** [comment]
Darn. No luck with that either. I added `console.log(this.locales);` before [the line throwing the error](https://github.com/shipshapecode/ember-flatpickr/blob/a28b6ab2246f1aa8d2651cc4425b201591c6983e/index.js#L10). It gets hit twice, the first time locales is an empty array and the second time it's undefined. I'll try [debugging ember build](https://cli.emberjs.com/release/advanced-use/debugging/#breakpoints) to see if I can get more info.

**Comment by rwwagner90** [comment]
@ppegusii do you have `ember-flatpickr` listed as a dependency or devDependency?

**Comment by ppegusii** [comment]
It's a dependency in my addon and not listed in package.json of the consuming app.

**Comment by rwwagner90** [comment]
Is your addon open source?

**Comment by rwwagner90** [comment]
Now that I think about it, if it's getting called twice, we could probably fix this for you by simply checking if `this.locales` exists and is an array before we call map on it.

**Comment by ppegusii** [comment]
Oh nice! I was thinking about doing that, but didn't know the consequences. I'll try that in my code and let you know if it works.

**Comment by ppegusii** [comment]
I changed [that line](https://github.com/shipshapecode/ember-flatpickr/blob/a28b6ab2246f1aa8d2651cc4425b201591c6983e/index.js#L10) to the following and the consuming project now builds and runs.
```
const localePaths = Object.prototype.toString.call(this.locales) === '[object Array]' ? this.locales.map(locale => `l10n/${locale}.js`) : [];
```

**Comment by rwwagner90** [comment]
@ppegusii does this PR fix your issue? https://github.com/shipshapecode/ember-flatpickr/pull/499

**Comment by ppegusii** [comment]
Sorry, build is failing with the [same error as the travis test](https://travis-ci.com/github/shipshapecode/ember-flatpickr/jobs/367317740#L642). I'm running node v12.16.1. I've never seen that syntax before.

**Comment by rwwagner90** [comment]
@ppegusii ah sorry, optional chaining and nullish coalescing seem to not be available until node 14. Let me refactor.

**Comment by rwwagner90** [comment]
@ppegusii okay, I refactored. Can you try https://github.com/shipshapecode/ember-flatpickr/pull/499 again please?

**Comment by ppegusii** [comment]
Yes! Many thanks @rwwagner90. Sorry for the delayed response.

**Comment by ppegusii** [comment]
@rwwagner90 do you know when this change will be released?

**Comment by rwwagner90** [comment]
@ppegusii just released 2.16.3

**Comment by ppegusii** [comment]
Thank you. You run a tight ship!
